src/testdir/test_quickfix.vim

```diff
@@ -6462,6 +6462,32 @@ func Test_quickfix_buffer_contents()
   call setqflist([], 'f')
 endfunc
 
+func Test_quickfix_update()
+  " Setup: populate a couple buffers
+  call setline(1, range(1,5))
+  let b1 = bufnr()
+  new
+  call setline(1, range(1,3))
+  let b2 = bufnr()
+  " Setup: set a quickfix list.
+  let items = [{'bufnr': b1, 'lnum': 1}, {'bufnr': b1, 'lnum': 2}, {'bufnr': b2, 'lnum': 1}, {'bufnr': b2, 'lnum': 2}]
+  call setqflist(items)
+
+  " Open the quickfix list, select the third entry.
+  copen
+  exe "normal jj\<CR>"
+  call assert_equal(3, getqflist({'idx' : 0}).idx)
+
+  " Update the quickfix list. Make sure the third entry is still selected.
+  call setqflist([], 'u', { 'items': items })
+  call assert_equal(3, getqflist({'idx' : 0}).idx)
+
+  " Update the quickfix list again, but this time with missing line number
+  " information. Confirm that we keep the current buffer selected.
+  call setqflist([{'bufnr': b1}, {'bufnr': b2}], 'u')
+  call assert_equal(2, getqflist({'idx' : 0}).idx)
```

Comment:
The test should clean up the newly created buffers.

Reply:
How do I ensure that happens even if the test fails?

Reply:
I've added a couple of `%bwipe!` commands at the end of the test, so I'm going to mark this done.

`<rant>`I skimmed the other tests in this file, and I'm not convinced they're being consistent about cleaning up the buffers they allocate. My inclination would be to try to solve this structurally rather than carefully: the test infrastructure/framework should make sure tests start with a clean slate, and furthermore, clean up any buffers/other stuff allocated during the test.`</rant>`

Reply:
I think one `%bwipe!` should be fine :)

Reply:
Oooh does that close all buffers? Ha! Got it.

Fixed.